pygmt/clib/conversion.py

```diff
@@ -304,7 +306,7 @@ def strings_to_ctypes_array(strings: Sequence[str]) -> ctp.Array:
     >>> [s.decode() for s in ctypes_array]
     ['first', 'second', 'third']
     """
-    return (ctp.c_char_p * len(strings))(*[s.encode() for s in strings])
+    return (ctp.c_char_p * len(strings))(*[s.encode() for s in np.asarray(strings)])
```

Comment:
With PR #3507, I believe we can revert changes in this file, and only let the function support a sequence of strings and a string dtype numpy array.

Reply:
Reverted in a927202acfcdb9a236a765534079bb821b798189